test/teststl.cpp

```diff
@@ -181,16 +183,16 @@ class TestStl : public TestFixture {
     struct CheckOptions
     {
         bool inconclusive = false;
-        Standards::cppstd_t cppstandard = Standards::CPPLatest;
+        const Settings* s = nullptr;
     };
 
 #define check(...) check_(__FILE__, __LINE__, __VA_ARGS__)
     template<size_t size>
     void check_(const char* file, int line, const char (&code)[size], const CheckOptions& options = make_default_obj()) {
-        const Settings settings1 = settingsBuilder(settings).certainty(Certainty::inconclusive, options.inconclusive).cpp(options.cppstandard).build();
+        const Settings* s = options.s ? options.s : (options.inconclusive ? &settings_i : &settings);
```

Comment:
`inconclusive` is not applied to custom settings but I think that is fine. We could add a sanity check (or overload) to make sure that nothing is supplied with the custom settings - but that applies across all tests and so is out-of-scope for this PR.